### PR TITLE
SNMP and sessionServiceType no longer supported

### PR DESCRIPTION
**Pull request by justinma246**
Ticket: #850 (https://github.com/kaazing/roadmap/issues/850)

An exception is now thrown when management.snmp is attempted to be used. Additionally, all files pertinent to SNMP has been removed from gateway.management

Ticket: #576 (https://github.com/kaazing/tickets/issues/576#event-705527611)

An exception also thrown when sessionServiceType is attempted to be used.

### CODE REVIEW

**Comment by dpwspoon** [comment]
What are all the management/bin files?

File: server/src/main/java/org/kaazing/gateway/server/config/parse/GatewayConfigParser.java [hunk 1]
```diff
@@ -158,16 +159,30 @@ private File getTranslatedConfigFile(final File configFile)
...
     }
+    
+    private void checkForOutdated(Element root) throws Exception {
+        Namespace namespace = root.getNamespace();
+        List<Element> children = root.getChildren("service", namespace);
+        for (Element child : children) {
+            Element typeChild = child.getChild("type", namespace);
+            String type = typeChild.getText();
+            if (type.equals("management.snmp")) {
+                throw new Exception("snmp management type is outdated and no longer supported.");
```
Comment (dpwspoon) [inline server/src/main/java/org/kaazing/gateway/server/config/parse/GatewayConfigParser.java]: change from outdated to just "is no longer supported", suggest renaming method name accordingly

**Comment by justinma246** [comment]
@dpwspoon renamed the method and changed the error message

**Comment by dpwspoon** [comment]
The automated builds failed with snmp related error

```
testServiceDefaultsWithoutSslCiphers(org.kaazing.gateway.management.impl.ServiceDefaultsConfigurationBeanImplTest) starting
the Mockery is not thread-safe: use a Synchroniser to ensure thread safety
the Mockery is not thread-safe: use a Synchroniser to ensure thread safety
ServiceDefaultsConfigurationBeanImplTest.testServiceDefaultsWithoutSslCiphers FAILED with exception java.lang.IllegalArgumentException: Unrecognized service type: management.snmp (0.58 secs)
java.lang.IllegalArgumentException: Unrecognized service type: management.snmp
	at org.kaazing.gateway.service.ServiceFactory.newService(ServiceFactory.java:47)
	at org.kaazing.gateway.server.context.resolve.GatewayContextResolver.resolveServices(GatewayContextResolver.java:488)
	at org.kaazing.gateway.server.context.resolve.GatewayContextResolver.resolve(GatewayContextResolver.java:236)
	at org.kaazing.gateway.server.test.Gateway.createGatewayContext(Gateway.java:128)
	at org.kaazing.gateway.server.test.Gateway.start(Gateway.java:97)
	at org.kaazing.gateway.management.impl.ServiceDefaultsConfigurationBeanImplTest.testServiceDefaultsWithoutSslCiphers(ServiceDefaultsConfigurationBeanImplTest.java:62)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:745)
```

Looks like there are tests you didn't remove, also I would recommend removing the files that have MIB in there name (they aren't jar files)

**Comment by justinma246** [comment]
@dpwspoon removed test (ServiceDefaultsConfigurationBeanImplTest.java) for snap and MIB-design.txt.
ran mvn clean install and gateway runs smoothly

**Pull request merged**